Include/pymacconfig.h

```diff
@@ -74,8 +77,14 @@
 #  define DOUBLE_IS_LITTLE_ENDIAN_IEEE754
 #endif
 
-#ifdef __i386__
+#if defined(__i386__) || defined(__x86_64__)
```

Comment:
Added ``__x86_64__`` to the check because the compiler included with Xcode 15.1 (and possibly earlier) no longer defines ``__i386__`` when building for Intel.